core/src/autogluon/core/metrics/__init__.py

```diff
@@ -167,9 +175,12 @@ def __call__(self, y_true, y_pred, sample_weight=None):
         if sample_weight is not None:
             return self._sign * self._score_func(y_true, y_pred,
                                                  sample_weight=sample_weight,
-                                                 **self._kwargs)
+                                                 **self._kwargs,
+                                                 **kwargs)
         else:
-            return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
+            return self._sign * self._score_func(y_true, y_pred,
+                                                 **self._kwargs,
+                                                 **kwargs)
```

Comment:
This might be problematic. For example, the keys in `kwargs` may duplicate the keys in `self._kwargs`. Thus, we may consider to merge these two dictionary and then feed into the `_score_func`. For example:

```python
kwargs.update(self._kwargs)
self._score_func(y_true, y_pred, **self._kwargs)
```

Reply:
It is unclear if this is what we want. It seems strange the user would specify a specific value in construction and then pass a different one at scoring time. We could definitely make it work, but it might be better to have it crash as it currently does. Any thoughts on if there is a valid use-case?

Reply:
I think users may want to overwrite some predefined kwargs.

Reply:
The code you provided does the opposite, it overrides the user kwargs with the predefined kwargs:

```
kwargs.update(self._kwargs)
self._score_func(y_true, y_pred, **kwargs)
```

Did you instead mean:

```
k = copy.deepcopy(self._kwargs)
k.update(kwargs)
self._score_func(y_true, y_pred, **k)
```

Reply:
Yeah, the 2nd version should be better.

Reply:
updated the PR to add support for the 2nd version + simplify the code and add documentation.